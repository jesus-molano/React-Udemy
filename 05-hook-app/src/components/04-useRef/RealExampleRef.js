import { useState } from 'react'
import MultipleCustomHooks from '../03-examples/MultipleCustomHooks'
import './useref.css'

const RealExampleRef = () => {

  const [show, setShow] = useState(false);

  const handleClick = () => {
    setShow(!show);
  }

  return (
    <div>
      <h1>Real Example useRef</h1>
      <hr />
      {show && <MultipleCustomHooks />}
      <button className="btn btn-outline-primary mt-3" onClick={handleClick}>Show</button>
    </div>
  )
}

export default RealExampleRef
