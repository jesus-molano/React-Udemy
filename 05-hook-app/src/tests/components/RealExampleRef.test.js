import { shallow } from 'enzyme'
import RealExampleRef from '../../components/04-useRef/RealExampleRef'

describe('RealExampleRef tests',() => {
  const wrapper = shallow(<RealExampleRef />);

  test('should show correctly', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);
  })
  
  test('should show correctly the component: <MultipleCustomHooks/>', () => {
    wrapper.find('button').simulate('click');
    expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
  })
  
})