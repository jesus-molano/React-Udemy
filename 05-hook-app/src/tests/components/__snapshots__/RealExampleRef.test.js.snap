// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RealExampleRef tests should show correctly 1`] = `
<div>
  <h1>
    Real Example useRef
  </h1>
  <hr />
  <button
    className="btn btn-outline-primary mt-3"
    onClick={[Function]}
  >
    Show
  </button>
</div>
`;
